src/main/java/duke/command/Storage.java

```diff
@@ -0,0 +1,143 @@
+package duke.command;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import duke.command.TaskList;
+import duke.task.Task;
+import duke.task.Todo;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.DukeException;
+
+public class Storage {
+
+    private File file;
+    TaskList taskList;
+    ArrayList<Task> list;
+
+    /**
+     * Constructs a storage object that takes in the file path as parameter.
+     * @param filePath The path at which the file is stored.
+     */
+    public Storage(String filePath) {
+        this.file = new File(filePath);
+        this.list = new ArrayList<Task>();
+    }
+
+    /**
+     * Loads the content in the file into the ArrayList of Task.
+     * @return The ArrayList of type Task which has been filled with the tasks from
+     *     the content of the file.
+     * @throws DukeException if file does not exist in the fiven file path.
+     * @throws FileNotFoundException if no file can be found in the given file path.
+     * @throws ParseException if format of the file does not fit the expected format.
+     * @throws IOException if file path cannot be found.
+     */
+    public ArrayList<Task> load() throws DukeException, FileNotFoundException, ParseException, IOException {
+        if (this.file.exists()) {
+            Scanner scannerTask = new Scanner(this.file);
+            while (scannerTask.hasNext()) {
+                String taskText = scannerTask.nextLine();
+                parseTextToTask(taskText, this.list);
+            }
+            return this.list;
+        } else {
+            file.createNewFile();
+            throw (new DukeException("File not found"));
+        }
+    }
+
+    /**
+     * Set the field of the ArrayList of task to the list of task from TaskList.
+     * @param taskList The TaskList from which the ArrayList of task is to be assigned
+     *                 to the class's ArrayList field.
+     */
+    public void setList(TaskList taskList) {
+        this.list = taskList.getList();
+    }
+
+    /**
+     * Parse the Text from the File to the corresponding Task objects.
+     * @param taskText the Line from the text to be parsed into Task object.
+     * @param list The list to which the resulting parsed text wish to be added.
+     * @throws ParseException if format of the text does not fit the expected format.
+     */
+    public void parseTextToTask(String taskText, ArrayList<Task> list) throws ParseException {
```

Comment:
Would Parser be a more appropriate class to handle the processing and adding of Tasks to the list in the program, while Storage only handles the operations on the external save file?